### PR TITLE
Refactor analyze_docker_image

### PR DESCRIPTION
**Pull request by PrajwalM2212**
Created analyze.py which contains all the functions
that analyze_docker_image did.
The analyze_docker_image function has
been refactored and divided into multiple functions

Resolves https://github.com/vmware/tern/issues/212

Signed-off-by: Prajwal M <prajwalmmath@gmail.com>

### CODE REVIEW

**Comment by PrajwalM2212** [comment]
@nishakm @rnjudge Can you review this ?

File: tern/report/analyze.py [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+def prepare_for_analysis(image_obj, dockerfile):
+    # find the layers that are imported
+    if dockerfile:
+        dhelper.set_imported_layers(image_obj)
+    # add notices for each layer if it is imported
+    image_setup(image_obj)
+
+
+def analyse_first_layer(binary, image_obj, shell, master_list, redo):
```
[the remark this answers is not in the document]
Comment (PrajwalM2212) [reply tern/report/analyze.py]: Done

**Comment by PrajwalM2212** [comment]
@nishakm After I rebased this, the tests failed as they could not find `stevedore` in report.py
Shouldn't we include https://github.com/openstack/stevedore in requirements.txt ?

**Comment by rnjudge** [comment]
@PrajwalM2212 I am very sorry but I accidentally updated the commit sha's on master hence the long list of conflicts for this PR. Could you please rebase this PR and fix the conflicts? We appreciate all the work that you have put it to this PR and others and I am so sorry to ask more work of you. Thank you in advance!

**Comment by PrajwalM2212** [comment]
I will update this later today.

**Pull request closed without merging**